### PR TITLE
Integration test setting and wiremock

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-765

### Change description ###

Integration test setting and wiremock

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -31,4 +31,11 @@
     <cve>CVE-2016-5394</cve>
     <cve>CVE-2016-6798</cve>
   </suppress>
+  <suppress>
```
Comment (nitinprabhu) [inline config/owasp/suppressions.xml]: could we please add a suppress until date depending when we think fix might be available ?

Comment (mdayican) [reply config/owasp/suppressions.xml]: added

File: config/owasp/suppressions.xml [hunk 2]
```diff
@@ -31,4 +31,11 @@
     <cve>CVE-2016-5394</cve>
     <cve>CVE-2016-6798</cve>
   </suppress>
+  <suppress  until="2020-06-01Z">
```
Comment (nitinprabhu) [inline config/owasp/suppressions.xml]: format is `<suppress until="2019-10-01">`

File: build.gradle [hunk 3]
```diff
@@ -209,7 +215,10 @@ dependencies {
   testCompile group: 'org.junit.platform', name: 'junit-platform-commons', version: versions.junitPlatform
   testCompile group: 'org.junit.platform', name: 'junit-platform-engine', version: versions.junitPlatform
 
-  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test'
+  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', withoutJunit4
+  testCompile group: 'com.github.tomakehurst', name: 'wiremock-jre8', version: '2.25.0', withoutJunit4
```
Comment (doncem) [inline build.gradle]: I believe presence of wiremock indicates that test using it is already integration test

File: build.gradle [hunk 4]
```diff
@@ -209,7 +215,10 @@ dependencies {
...
+
+  integrationTestCompile group: 'org.springframework.cloud', name: 'spring-cloud-contract-wiremock', version: '2.1.3.RELEASE', withoutMockitoStandAlone
```
Comment (doncem) [inline build.gradle]: this will make 2 types of wiremock present 🤔

File: config/owasp/suppressions.xml [hunk 5]
```diff
@@ -31,4 +31,11 @@
     <cve>CVE-2016-5394</cve>
     <cve>CVE-2016-6798</cve>
   </suppress>
+  <suppress  until="2020-06-01">
```
Comment (nitinprabhu) [inline config/owasp/suppressions.xml]: although this should be set not more than a month but can be changed later as this Pr needs to go

File: build.gradle (hunk 3 above)
Comment (mdayican) [reply build.gradle]: it will be used in integration test.

File: build.gradle (hunk 4 above)
Comment (mdayican) [reply build.gradle]: we exclude the one comes from spring-cloud-contract-wiremock and it should 1 in classpath

**Comment by mdayican** [comment]
> owasp

it is excluded for  owasp check as the other version brings too much vulnerabilities. before using this version  used suppressions  and it is advised by @nitinprabhu to use another jar instead of suppressing vulnerabilities.

File: build.gradle (hunk 3 above)
Comment (doncem) [reply build.gradle]: hence wrongly assigned - in `test` scope, not `integration`

File: build.gradle (hunk 4 above)
Comment (doncem) [reply build.gradle]: ah there's `standalone` and `jre8` dependencies

Comment (mdayican) [reply build.gradle]: next PR I will make it integration scope

**Pull request merged**